src/ngx_http_lua_subrequest.c

```diff
@@ -382,15 +404,68 @@ ngx_http_lua_ngx_location_capture_multi(lua_State *L)
 
             type = lua_type(L, -1);
 
-            if (type == LUA_TNIL) {
+            switch (type) {
+            case LUA_TNIL:
                 method = NGX_HTTP_GET;
+                break;
 
-            } else {
-                if (type != LUA_TNUMBER) {
+            case LUA_TNUMBER:
+                method = (ngx_uint_t) lua_tonumber(L, -1);
+                break;
+
+            case LUA_TSTRING:
+                lua_method_name = (u_char *) lua_tolstring(L, -1,
+                                                           &method_name_length);
+                if (method_name_length == 0) {
                     return luaL_error(L, "Bad http request method");
                 }
+                
+                methods_number = sizeof(ngx_http_lua_ordered_methods) /
+                    sizeof(ngx_http_lua_ordered_methods[0]);
+
+                for (method_index = 0; method_index < methods_number;
```

Comment:
Hmm, I'm afraid I don't quite like this `O(n)` operation on such a hot code path. Maybe introduce a hash table here?